src/vivarium_conic_lsff/results_processing/process_results.py

```diff
@@ -0,0 +1,228 @@
+            .reset_index(drop=True)
+            # TODO: add back when we have scenarios
+            # .rename(columns={project_globals.OUTPUT_SCENARIO_COLUMN: SCENARIO_COLUMN}))
+            )
+    data[project_globals.INPUT_DRAW_COLUMN] = data[project_globals.INPUT_DRAW_COLUMN].astype(int)
+    data[project_globals.RANDOM_SEED_COLUMN] = data[project_globals.RANDOM_SEED_COLUMN].astype(int)
+    with (path.parent / 'keyspace.yaml').open() as f:
+        keyspace = yaml.full_load(f)
+    return data, keyspace
+
+
+def filter_out_incomplete(data, keyspace):
+    output = []
+    for draw in keyspace[project_globals.INPUT_DRAW_COLUMN]:
+        # For each draw, gather all random seeds completed for all scenarios.
+        random_seeds = set(keyspace[project_globals.RANDOM_SEED_COLUMN])
+        draw_data = data.loc[data[project_globals.INPUT_DRAW_COLUMN] == draw]
+        # TODO: add back when we have scenarios
+        # for scenario in keyspace[project_globals.OUTPUT_SCENARIO_COLUMN]:
+        #     seeds_in_data = draw_data.loc[data[SCENARIO_COLUMN] == scenario,
+        #                                   project_globals.RANDOM_SEED_COLUMN].unique()
+        #     random_seeds = random_seeds.intersection(seeds_in_data)
+        draw_data = draw_data.loc[draw_data[project_globals.RANDOM_SEED_COLUMN].isin(random_seeds)]
+        output.append(draw_data)
+    return pd.concat(output, ignore_index=True).reset_index(drop=True)
+
+
+def aggregate_over_seed(data):
+    non_count_columns = []
+    for non_count_template in project_globals.NON_COUNT_TEMPLATES:
+        non_count_columns += project_globals.RESULT_COLUMNS(non_count_template)
+    count_columns = [c for c in data.columns if c not in non_count_columns + GROUPBY_COLUMNS]
+
+    # non_count_data = data[non_count_columns + GROUPBY_COLUMNS].groupby(GROUPBY_COLUMNS).mean()
+    count_data = data[count_columns + GROUPBY_COLUMNS].groupby(GROUPBY_COLUMNS).sum()
+    return pd.concat([
+        count_data,
+        # non_count_data
+    ], axis=1).reset_index()
+
+
+def pivot_data(data):
+    return (data
+            .set_index(GROUPBY_COLUMNS)
+            .stack()
+            .reset_index()
+            .rename(columns={f'level_{len(GROUPBY_COLUMNS)}': 'process', 0: 'value'}))
+
+
+def sort_data(data):
+    sort_order = [
+        'age_group',
+        'risk',
+        'cause',
+        'treatment_group',
+        'measure',
+        'input_draw'
+    ]
+    sort_order = [c for c in sort_order if c in data.columns]
+    other_cols = [c for c in data.columns if c not in sort_order]
+    data = data[sort_order + other_cols].sort_values(sort_order)
+    return data.reset_index(drop=True)
+
+
+def split_processing_column(data, with_cause, state, transition, stratified):
+    data['treatment_group'] = 'all'
+    for level in stratified.levels:
+        data['process'], data[level[0]] = data.process.str.split(level[1]).str
```

Comment:
This is fragile, since the order of iteration over a dictionary's keys is not guaranteed, and the order you split the string by matters. It has to iterate through the keys right to left. 

I wrote my `split_processing_column` function specifically for the template strings I have in Zenon. If we're trying to write a more generalizable function, this is not a good pattern because it's so dependent on the order of the strings.